qiskit_optimization/algorithms/qrao/quantum_random_access_encoding.py

```diff
@@ -0,0 +1,579 @@
+        return self._var2op
+
+    @property
+    def q2vars(self) -> list[list[int]]:
+        """Each element contains the list of decision variable indices encoded on that qubit"""
+        return self._q2vars
+
+    @property
+    def compression_ratio(self) -> float:
+        """Compression ratio. Number of decision variables divided by number of qubits"""
+        return self.num_vars / self.num_qubits
+
+    @property
+    def minimum_recovery_probability(self) -> float:
+        """Minimum recovery probability, as set by ``max_vars_per_qubit``"""
+        n = self.max_vars_per_qubit
+        return (1 + 1 / np.sqrt(n)) / 2
+
+    @property
+    def qubit_op(self) -> SparsePauliOp:
+        """Relaxed Hamiltonian operator.
+
+        Raises:
+            RuntimeError: If the objective function has not been set yet. Use the ``encode`` method
+                to construct the Hamiltonian, or make sure that the objective function has been set.
+        """
+        if self._qubit_op is None:
+            raise RuntimeError(
+                "Cannot return the relaxed Hamiltonian operator: no objective function has been "
+                "provided yet. Use the ``encode`` method to construct the Hamiltonian, or make "
+                "sure that the objective function has been set."
+            )
+        return self._qubit_op
+
+    @property
+    def offset(self) -> float:
+        """Relaxed Hamiltonian offset
+
+        Raises:
+            RuntimeError: If the offset has not been set yet. Use the ``encode`` method to construct
+                the Hamiltonian, or make sure that the objective function has been set.
+        """
+        if self._offset is None:
+            raise RuntimeError(
+                "Cannot return the relaxed Hamiltonian offset: The offset attribute cannot be "
+                "accessed until the ``encode`` method has been called to generate the qubit "
+                "Hamiltonian. Please call ``encode`` first."
+            )
+        return self._offset
+
+    @property
+    def problem(self) -> QuadraticProgram:
+        """The ``QuadraticProgram``  encoding a QUBO optimization problem
+
+        Raises:
+            RuntimeError: If the ``QuadraticProgram`` has not been set yet. Use the ``encode``
+                method to set the problem.
+        """
+        if self._problem is None:
+            raise RuntimeError(
+                "This object has not been associated with a ``QuadraticProgram``. "
+                "Please use the ``encode`` method to set the problem."
+            )
+        return self._problem
+
+    def freeze(self):
+        """Freeze the object to prevent further modification.
+
+        Once an instance of this class is frozen, ``encode`` can no longer be called.
+        """
+        if not self._frozen:
+            self._qubit_op = self._qubit_op.simplify()
+            self._qubit_op = PauliSumOp(self._qubit_op)
```

Comment:
I think for new code we should be avoiding function, like PauliSumOp which is already deprecated and use SparsePauliOp directly. Maybe this is needed in order to fit into the rest of the optimization framework as PauliSumOp use there is deprecated and goes away. Not sure - I did not look further - it would be nice if this new code did not use it unless it really has to,

Reply:
Right, I will remove `PauliSumOp`. (Actually, it works without using `PauliSumOp`)